lib/saml_idp/xml_security.rb

```diff
@@ -43,12 +41,12 @@ def initialize(msg = nil, error_code = nil)
 
       C14N = 'http://www.w3.org/2001/10/xml-exc-c14n#'
       DSIG = 'http://www.w3.org/2000/09/xmldsig#'
+      DS_NS = { "ds" => DSIG }
 
-      attr_accessor :signed_element_id
+      attr_accessor :document
 
       def initialize(response)
-        super(response)
-        extract_signed_element_id
+        @document = Nokogiri.parse(response)
```

Comment:
recommendation: Use `Nokogiri.XML` instead of `Nokogiri.parse` to prevent flaky tests down the road as `Nokogiri.parse` assumes that if the input is HTML when the input is a URL/Pathname object (i.e., some IO object that can be "read"). 

Fortunately, the method signature is the same and shouldn't require any additional arguments.

Reply:
thanks! updated in https://github.com/18F/saml_idp/pull/110/commits/512ad9b1609884781034773f6c4ccd53e7036a82